lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -1,127 +1,119 @@
 
-        'drb.v4.run.2020-08,drb.v4.run.2020-09,drb.v4.run.2020-10,'
-
-    This code is adapted from Javascript in the Pbench dashboard's
-    'moment_constants.js' getAllMonthsWithinRange function.
+    TODO: when we get our user model straightened out, this will
+    convert the external user representation (either username or email
+    address) to whatever internal representation we want to store in
+    the Elasticsearch database, whether that's a SQL row ID or something
+    else. For now, it just validates that the username string is a real
+    user.
 
     Args:
-        prefix ([string]): The Pbench server index prefix
-        index ([string]): The desired monthly index root
-        start ([datetime.datetime]): The start time
-        end ([datetime.datetime]): The end time
+        value: external user representation
 
     Returns:
-        [string]: A comma-separated list of month-qualified index names
+        internal username representation
     """
-    monthResults = list()
-    queryString = ""
-    for m in rrule.rrule(rrule.MONTHLY, dtstart=start, until=end):
-        monthResults.append(m.strftime("%Y-%m"))
-
-    # TODO: hardcoding the index here is risky. We need a framework to
-    # help the web services understand index versions and template
-    # formats, probably by building a persistent database from the
-    # index template documents at startup. This is TBD.
-    for monthValue in monthResults:
-        if index == "v4.result-data.":
-            queryString += f"{prefix + index + monthValue}-*,"
-        else:
-            queryString += f"{prefix + index + monthValue},"
-    return queryString
-
+    try:
+        return Auth.validate_user(value)
+    except Exception:
+        raise ConversionError(value, "username", type(value).__name__)
 
-def convert_date(value: str):
-    return parser.parse(value)
 
+def convert_json(value: dict) -> dict:
+    """
+    convert_json Process a parameter of JSON type; currently just by validating
+    that it's a Python dict.
 
-def convert_username(value: str):
-    return Auth.validate_user(value)
+    Args:
+        value: JSON dict
 
+    Raises:
+        TypeError: The value is not a dict
 
-def convert_json(value: dict):
+    Returns:
+        The JSON dict
+    """
     if type(value) is not dict:
-        raise TypeError(f"Unexpected type {type(value).__name__}")
+        raise ConversionError(value, dict.__name__, type(value).__name__)
     return value
 
 
-def convert_string(value: str):
+def convert_string(value: str) -> str:
+    """
+    convert_string Verify that the parameter value is a string (e.g.,
+    not a JSON dict, or an int), and return it.
+
+    Args:
+        value: parameter value
+
+    Raises:
+        TypeError: the value is not a string
+
+    Returns:
+        the input value
+    """
+    if type(value) is not str:
+        raise ConversionError(value, str.__name__, type(value).__name__)
     return str(value)
```

Comment:
At line 117, we know that `value` is of type `str`...so why not just return it?  (I think this operation is returning a _copy_ of it -- that seems undesirable.)

Reply:
You're right -- I should have removed the typecast when I added the explicit type check.